resources/definitions/tenlog_printer.def.json

```
@@ -0,0 +1,295 @@
+
+        "line_width": { "value": "machine_nozzle_size" },
+
+        "optimize_wall_printing_order": { "value": "True" },
+
+        "material_initial_print_temperature": { "value": "material_print_temperature" },
+        "material_final_print_temperature": { "value": "material_print_temperature" },
+        "material_flow": { "value": 100 },
+        "travel_compensate_overlapping_walls_0_enabled": { "value": "False" },
+
+        "z_seam_type": { "value": "'back'" },
+        "z_seam_corner": { "value": "'z_seam_corner_weighted'" },
+        
+        "infill_sparse_density": { "value": "20" },
+        "infill_pattern": { "value": "'lines'" },
+
+        "infill_overlap": { "value": 30.0 },
+        "infill_wipe_dist": { "value": 0.0 },
+        "wall_0_wipe_dist": { "value": 0.0 },
+
+        "fill_perimeter_gaps": { "value": "'everywhere'" },
+        "fill_outline_gaps": { "value": false },
+        "filter_out_tiny_gaps": { "value": false },
+
+        "retraction_speed": {
+            "maximum_value_warning": "machine_max_feedrate_e if retraction_enable else float('inf')",
+            "maximum_value": 200
+        },
+        "retraction_retract_speed": {
+            "maximum_value_warning": "machine_max_feedrate_e if retraction_enable else float('inf')",
+            "maximum_value": 200
+        },
+        "retraction_prime_speed": {
+            "maximum_value_warning": "machine_max_feedrate_e if retraction_enable else float('inf')",
+            "maximum_value": 200
+        },
+        
+        "retraction_hop_enabled": { "value": "False" },
+        "retraction_hop": { "value": 0.2 },
+        "retraction_combing": { "value": "'off' if retraction_hop_enabled else 'noskin'" },
+        "retraction_combing_max_distance": { "value": 0 },
+        "travel_avoid_other_parts": { "value": false },
+        "travel_avoid_supports": { "value": true },
+        "travel_retract_before_outer_wall": { "value": true },
+        
+	"retraction_hop_after_extruder_switch_height": { "value": 0.0 },
+
+        "retraction_enable": { "value": true },
+        "retraction_count_max": { "value": 100 },
+        "retraction_extrusion_window": { "value": 4 },
+        "retraction_min_travel": { "value": 1.5 },
+
+        "cool_fan_full_at_height": { "value": "layer_height_0 + 2 * layer_height" },
+        "cool_fan_enabled": { "value": true },
+        "cool_min_layer_time": { "value": 10 },
+
+        "adhesion_type": { "value": "'skirt'" },
+        "brim_replaces_support": { "value": true },
+        "skirt_gap": { "value": 3.0 },
+        "skirt_line_count": { "value": 2 },
+	"raft_margin":{"default_value": 2 },
+	"raft_airgap":	{ "default_value": 0.2 },
+
+	"switch_extruder_retraction_amount":	{ "value": 8.0 },
+	"switch_extruder_retraction_speeds":	{ "default_value": 60.0 },
+	"prime_tower_wipe_enabled":	{ "default_value": false },
+	"prime_tower_brim_enable":	{ "default_value": true },
```

Mix of tabs and spaces
```suggestion
        "raft_margin":{"default_value": 2 },
        "raft_airgap":	{ "default_value": 0.2 },

        "switch_extruder_retraction_amount":	{ "value": 8.0 },
        "switch_extruder_retraction_speeds":	{ "default_value": 60.0 },
        "prime_tower_wipe_enabled":	{ "default_value": false },
        "prime_tower_brim_enable":	{ "default_value": true },
```